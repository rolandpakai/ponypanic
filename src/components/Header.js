import { useContext } from 'react';
import { ThemeContext } from '../contexts/ThemeContext';

const Header = () => {
  const { theme } = useContext(ThemeContext);

  return (
    <header className="header">
      <div className="header-logo">
        <span className="header-span">
          <img 
            src={`./themes/${theme}/title.png`}
            className="header-img"
            alt=""
          />
        </span>
      </div>
    </header>
  )
}

export default Header;