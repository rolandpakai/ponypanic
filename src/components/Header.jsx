import React, { useContext } from "react";
import Box from "@mui/material/Box";
import Container from "@mui/material/Container";
import Stack from "@mui/material/Stack";
import { ThemeContext } from "../contexts/ThemeContext";

const Header = () => {
  const { theme } = useContext(ThemeContext);

  return (
    <Box>
      <Container maxWidth="md">
        <Stack
          direction="row"
          justifyContent="center"
          maxWidth="xs"
          sx={{
            padding: "2rem",
          }}
        >
          <img
            alt="title"
            className="header-img"
            src={`./themes/${theme}/title.png`}
            style={{
              display: "block",
              height: "110px",
              objectFit: "contain",
            }}
          />
        </Stack>
      </Container>
    </Box>
  );
};

export default Header;
