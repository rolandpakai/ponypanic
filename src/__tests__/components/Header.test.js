import { render, act } from '@testing-library/react';
import Header from '../../components/Header';

describe('Header Component', () => {
  test('Header renders', () => {
    const {container} = render(<Header />);
    
		expect(container).toMatchSnapshot();
  });
});