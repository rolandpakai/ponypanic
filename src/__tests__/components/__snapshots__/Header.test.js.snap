// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Header renders 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
    >
      <div
        class="css-1hfk0uz-MuiStack-root"
      >
        <img
          alt="title"
          class="header-img"
          src="./themes/pony/title.png"
          style="display: block; height: 110px; object-fit: contain;"
        />
      </div>
    </div>
  </div>
</div>
`;
